### DISCUSSION TITLE
Help with C indentation

### DISCUSSION THREAD

**Discussion by ldelossa** [helix-editor/helix#9730] [Q&A] [answered]
Hello. 

For my day job I use a tab width of 8 for the C portion of our code. 

My code base is using this indentation already and I use this Vim configuration to get the correct indentation:
`autocmd FileType c,cpp setlocal tabstop=8 shiftwidth=8`

Now, I've been trying to get `helix` to do the same thing when C file are opened. I tried a few way but I ultimately winded up with this:

```
[[language]]
name = "c"
scope = "source.c"
injection-regex = "c"
file-types = ["c"]
roots = []
auto-format = true
comment-token = "//"
indent = { tab-width = 8, unit = "    " }

[[language]]
name = "cpp"
scope = "source.cpp"
injection-regex = "cpp"
file-types = ["cpp", "cc", "cxx", "hpp", "hxx", "hh"]
roots = []
auto-format = true
comment-token = "//"
indent = { tab-width = 8, unit = "    " }
``` 

When I open up C files tho, 4 space tab are still present. 

From searching through issues I keep seeing that Helix will try to guess the indentation from the file, but I don't see how this can work if we can't set the tab width before this "guess" is made? Maybe I'm just configuring something wrong here?

**Comment by kirawi** [comment, marked as answer]
Set unit to \t

**Reply by ldelossa** [reply]
Hmm, still didn't seem to work for me:

![image](https://github.com/helix-editor/helix/assets/5642902/e2ac7c27-c312-423c-9e2a-22a4544c0a5f)

**Reply by ldelossa** [reply]
Oh okay I also needed to do ":indent-style t"
